File: exercises/day11-AWorldWithoutMock/java/src/main/java/service/ToyProductionService.java
package service;

import domain.Toy;
import domain.ToyRepository;

public class ToyProductionService {

    private final ToyRepository repository;
    private final NotificationService notificationService;

    public ToyProductionService(ToyRepository repository,
                                NotificationService notificationService) {
        this.repository = repository;
        this.notificationService = notificationService;
    }

    public void assignToyToElf(String toyName) {
        Toy toy = repository.findByName(toyName);
        if (toy != null && toy.getState() == Toy.State.UNASSIGNED) {
            toy.setState(Toy.State.IN_PRODUCTION);
            repository.save(toy);
            notificationService.notifyToyAssigned(toy);
        }
    }
}

File: exercises/day11-AWorldWithoutMock/kotlin/src/main/kotlin/service/ToyProductionService.kt
package service

import domain.Toy
import domain.ToyRepository

class ToyProductionService(
    private val repository: ToyRepository,
    private val notificationService: NotificationService
) {
    fun assignToyToElf(toyName: String) {
        val toy = repository.findByName(toyName)
        if (toy != null && toy.state == Toy.State.UNASSIGNED) {
            toy.state = Toy.State.IN_PRODUCTION
            repository.save(toy)
            notificationService.notifyToyAssigned(toy)
        }
    }
}
